text/0000-increase-build-plan-flexibility.md

```diff
@@ -0,0 +1,121 @@
+
+# What it is
+[what-it-is]: #what-it-is
+
+The `version` key should be removed as a top-level entry in both the build plan and buildpack plan.  Buildpacks that need to coordinate versions from other parties should use an agreed upon metadata key (the conventional, but not required, choice is `version`) to replicate their existing functionality.  The BOM schema will be updated to remove `version` as a top-level key as well, to match its relation to the buildpack plan.
+
+The `lifecycle` is required to implement temporary compatibility with the previous build plan and buildpack plan structure by converting the `version` key in `metadata` into and out of a top-level key at the appropriate locations.  If both a `version` metadata entry and the top-level `version` key are both set, the lifecycle should fail with a descriptive error.
+
+# How it Works
+[how-it-works]: #how-it-works
+
+N/A
+
+# Drawbacks
+[drawbacks]: #drawbacks
+
+The primary drawback to this RFC is that it is a breaking API change.
+
+The secondary drawback is that without a top-level key, buildpack authors that would like collaborate on versions would need to read documentation from the buildpack they wish to collaborate with to determine the name of the key used for collaboration.  Given that that author would need to read the buildpack's documentation to determine if it honored the top-level `version` key today, it's unclear if this is a significant change.
+
+# Alternatives
+[alternatives]: #alternatives
+
+This RFC is a competing proposal to the [Build Plan Merging RFC](https://github.com/buildpacks/rfcs/pull/67).  That RFC's proposal (as it stands today) is directionally different as its goal is to ensure clarity between collaborators by increasing the amount of structure in the build plan and buildpack plan.  The choice for which one of these two proposals is "best" depends directly on your view of the usage of the plans.
+
+The impact of not making this change is negligible as many buildpacks simply ignore the key completely.
+
+# Prior Art
+[prior-art]: #prior-art
+
+A non-exhaustive list of buildpacks from the Paketo project, the riff project, and distributed commercially by VMware (Tanzu-branded) represents a cross-section of buildpack implementations.
+
+Uses `version`:
+
+  * [`paketo-buildpacks/dotnet-core-aspnet`](https://github.com/paketo-buildpacks/dotnet-core-aspnet)
+  * [`paketo-buildpacks/dotnet-core-runtime`](https://github.com/paketo-buildpacks/dotnet-core-runtime)
+  * [`paketo-buildpacks/dotnet-core-sdk`](https://github.com/paketo-buildpacks/dotnet-core-sdk)
+  * [`paketo-buildpacks/go-compiler`](https://github.com/paketo-buildpacks/go-compiler)
+  * [`paketo-buildpacks/httpd`](https://github.com/paketo-buildpacks/httpd)
+  * [`paketo-buildpacks/nginx`](https://github.com/paketo-buildpacks/nginx)
+  * [`paketo-buildpacks/node-engine`](https://github.com/paketo-buildpacks/procfile)
```

Comment:
This link has the incorrect url target

Reply:
@nebhale I didn't realize your procfile buildpack installed node too :P